base/dict.jl

```diff
@@ -189,6 +189,12 @@ function grow_to!{K,V}(dest::Associative{K,V}, itr, st)
     return dest
 end
 
+
+"""
+    similar(d::Dict)
+
+Create an empty dictionary with the same key and value types as `d`.
+"""
 similar{K,V}(d::Dict{K,V}) = Dict{K,V}()
 similar{K,V}(d::Dict, ::Type{Pair{K,V}}) = Dict{K,V}()
```

Comment:
The second method allows you to build a `Dict` with a different type for the keys and or values. It might not seem too useful, but it can be used when writing generic code that works with `Dict`s and another type of `Associative` with custom behavior, for example.